aderyn_core/src/context/browser/extractor.rs

```diff
@@ -125,3 +126,62 @@ impl ASTConstVisitor for ExtractReferencedDeclarations {
         Ok(true)
     }
 }
+
+// Extract Reference Declaration IDs
+pub struct ExtractReferencedDeclarationsConditionally<'a> {
```

Comment:
@TilakMaddy do you expect this to be used elsewhere?

Reply:
Nothing in my mind right now, but intuitively it feels like a useful utility :)